### DISCUSSION TITLE
Should we add getters for segments?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#34] [Ideas]
https://github.com/sablierhq/v2-core/pull/31 implemented getters for all stream struct members, except the complex segment-related arrays in the `SablierV2Pro` contract.

Should we add getters for `segmentAmounts`, `segmentExponents` and `segmentMilestones`?

**Comment by razgraf** [comment]
I'd be pro doing that, as you can't really read struct data in solidity without specific getters. If it doesn't make the contract too big, it would be a nice DX feature.

**Reply by PaulRBerg** [reply]
You actually can read struct data if the mapping is declared as public. Ours isn't though.

And you don't really need the specific getters, you could get the segments from the `getStream` function.

It's more a question about whether anyone at all would need just the segments and not the other stream data .. and I think that the answer is "yes". There might be some cases when the frontend might need access to those variables - e.g. if we implement https://github.com/sablierhq/v2-core/issues/16, we need the segment data in the frontend to be able to provide a guess.

**Comment by PaulRBerg** [comment]
Locking, we ended up adding getters for segments:

https://github.com/sablierhq/v2-core/blob/fcae94c7d79c90d3b9548a1777366f932c47c054/src/interfaces/ISablierV2Pro.sol#L18-L31

**Reply by razgraf** [reply]
Why not `getSegment(uint index)` too?
